axelrod/fingerprint.py

```diff
@@ -0,0 +1,267 @@
+        """
+        edge_scores = [np.mean([compute_final_score_per_turn(scores)[0] for scores
+                                in interactions[edge]]) for edge in edges]
+        point_scores = dict(zip(points, edge_scores))
+        return point_scores
+
+    def fingerprint(self, turns=50, repetitions=10, step=0.01, processes=None,
+                    filename=None, in_memory=False, progress_bar=True):
+        """Build and play the spatial tournament.
+
+        Creates the probes and their edges then builds a spatial tournament
+        where the original strategy only plays probes whose coordinates sum to
+        less than 1 (or equal). Probes whose coordinates sum to more than 1 play
+        the Dual Strategy.
+
+        Parameters
+        ----------
+        turns : integer, optional
+            The number of turns per match
+        repetitions : integer, optional
+            The number of times the round robin should be repeated
+        step : float, optional
+            The separation between each Point. Smaller steps will
+            produce more Points that will be closer together.
+        processes : integer, optional
+            The number of processes to be used for parallel processing
+        progress_bar : bool
+            Whether or not to create a progress bar which will be updated
+
+        Returns
+        ----------
+        self.data : dictionary
+            A dictionary where the keys are coordinates of the form (x, y) and
+            the values are the mean score for the corresponding interactions.
+        """
+
+        if on_windows and (filename is None):
+            in_memory = True
+        elif filename is not None:
+            outputfile = open(filename, 'w')
+            filename = outputfile.name
+        else:
+            outputfile = NamedTemporaryFile(mode='w')
+            filename = outputfile.name
+
+        edges, tourn_players = self.construct_tournament_elements(step)
+        self.step = step
+        self.spatial_tournament = axl.SpatialTournament(tourn_players,
+                                                        turns=turns,
+                                                        repetitions=repetitions,
+                                                        edges=edges)
+        self.spatial_tournament.play(build_results=False,
+                                     filename=filename,
+                                     processes=processes,
+                                     in_memory=in_memory,
+                                     progress_bar=progress_bar)
+        if in_memory:
+            self.interactions = self.spatial_tournament.interactions_dict
+        else:
+            self.interactions = read_interactions_from_file(filename)
+
+        self.data = self.generate_data(self.interactions, self.points, edges)
+        return self.data
+
+    def plot(self, col_map='seismic', interp_method='none'):
```

Comment:
The tiniest of things (not a blocker for me). Do we want `interp_method` to be `interpolation` so that's it's the same variable name as for `imshow`?

Not a blocker for me, happy to let @meatballs and/or @marcharper have the final say on that.